### PR TITLE
[react-native-screens] update to 3.7.0

### PR DESCRIPTION
**Pull request by cruzach**
# Why

https://linear.app/expo/issue/ENG-1889/upgrade-react-native-screens

# How

`et uvm`, also had to do some manual fixes:

- adding new xml files
- add `host.exp.expoview.R` and `host.exp.expoview.BuildConfig` imports
- fixed error `ScreenStackHeaderConfig.kt: (374, 65): Type mismatch: inferred type is Context? but Context was expected` by passing `Context`
- 
# Test Plan

NCL

Out of curiosity- what purpose do the red and black Views serve on the nativeStack screen?
Screen Shot 2021-09-07 at 4 37 35 PM

### CODE REVIEW

File: ios/vendored/unversioned/react-native-screens/ios/RNSScreenWindowTraits.m [hunk 1]
```diff
@@ -21,20 +22,21 @@ + (void)assertViewControllerBasedStatusBarAppearenceSet
...
-      if (firstWindow != nil) {
-        [[firstWindow rootViewController] setNeedsStatusBarAppearanceUpdate];
-      }
-    } else
+                     if (@available(iOS 13, *)) {
+                       UIWindow *firstWindow = [[[UIApplication sharedApplication] windows] firstObject];
+                       if (firstWindow != nil) {
+                         [[firstWindow rootViewController] setNeedsStatusBarAppearanceUpdate];
+                       }
+                     } else
```
Comment (tsapeta) [inline ios/vendored/unversioned/react-native-screens/ios/RNSScreenWindowTraits.m:34]: 😱

**Review by tsapeta — APPROVED** [review]

**Comment by tsapeta** [comment]
> Out of curiosity- what purpose do the red and black Views serve on the nativeStack screen?

To be honest — I don't know 😅 It was copied over from the upstream repo, but I think we should maybe update this demo a little bit and even add more examples 🤔 This one is very basic and doesn't cover many features I guess

**Comment by cruzach** [comment]
I think to update the example, we'll have to also update react-navigation from v4 to v6 in NCL (good idea anyways)

I can do that as part of this PR

File: ios/vendored/unversioned/react-native-screens/ios/RNSScreenWindowTraits.m (hunk 1 above)
Comment (cruzach) [reply ios/vendored/unversioned/react-native-screens/ios/RNSScreenWindowTraits.m:34]: ewww...

Running Re-indent over the new files now

**Comment by cruzach** [comment]
> I think to update the example, we'll have to also update react-navigation from v4 to v6 in NCL (good idea anyways)
I can do that as part of this PR

On second thought, this upgrade would have quite a few changes involved so just in case we want to revert either this PR or the react-navigation upgrade in NCL, I'm going to land the latter in a separate PR

**Pull request merged**